### PR TITLE
feat: common keystore utility

### PR DESCRIPTION
**Pull request by sherpalden**
This feature can be used to encrypt the private key of chains using `aes-256-gcm` algorithm with the passphrase so that the extra layer of security is added for the keystore. 

Some chains like ICON and EVM chains inherently have this feature in their CLI tools. But chains like SUI doesn't have it. For such chains this feature is useful.

### CODE REVIEW

File: relayer/chains/evm/keys.go [hunk 1]
```diff
@@ -93,3 +94,7 @@ func (p *Provider) ImportKeystore(ctx context.Context, keyPath, passphrase strin
 func (p *Provider) keystorePath(addr string) string {
 	return path.Join(p.cfg.HomeDir, "keystore", p.NID(), addr)
 }
+
+func (p *Provider) ConvertPrivateKey(ctx context.Context, keyPath, passphrase string) (string, error) {
+	return "", fmt.Errorf("not required for evm chains")
```
Comment (bcsainju) [inline relayer/chains/evm/keys.go:99]: would "applicable" sound better than "required" ?

File: cmd/keystore.go [hunk 2]
```diff
@@ -160,6 +167,36 @@ func (k *keystoreState) importKey(a *appState) *cobra.Command {
 	return importCmd
 }
 
+// converts private key to keystore
+func (k *keystoreState) convertKey(a *appState) *cobra.Command {
+	convertCmd := &cobra.Command{
+		Use:   "convert",
+		Short: "convert keystore",
+		RunE: func(cmd *cobra.Command, args []string) error {
```
Comment (bcsainju) [inline cmd/keystore.go:175]: Can you add example as well ? that would make the usage more clear.

File: relayer/chains/evm/keys.go (hunk 1 above)
Comment (sherpalden) [reply relayer/chains/evm/keys.go]: done

File: cmd/keystore.go (hunk 2 above)
Comment (sherpalden) [reply cmd/keystore.go:176]: done

**Review by bcsainju — APPROVED** [review]

**Comment by debendraoli** [comment]
I may no need convert cmd, this is the against the original idea, we don't want any residual keys living on relayer.

I believe, this can be pretty much done with ease without such complexity.

**Review by debendraoli — CHANGES_REQUESTED** [review]
We may not need convert cmd, this is the against the original idea, we don't want any residual keys living on relayer.

I believe, this can be pretty much done with ease without such complexity.

**Comment by sherpalden** [comment]
not required...

**Pull request closed without merging**